### PR TITLE
Update ExtensionPage Style

### PR DESCRIPTION
**Pull request by CDK2020**
# EDIT: Please see [this comment](https://github.com/flarum/core/pull/1560#issuecomment-416083728) for the most updated info

#### Changes proposed in this pull request:
Make the ExtensionsPage 2 column and hide the description on mobile

#### Screenshots:

**Desktop:**

Before:

image

After:
image

**Mobile:**

Before:

image

After:

image

### CODE REVIEW

**Comment by ardacebi** [comment]
Thank you so much for this @CDK2020, especially for the 2 column feature. :)

**Comment by gurjyot** [comment]
Actually, having a description for extensions in mobile wasn't bad. I would rather suggest providing a drop-down description in mobile view if you wanna hide it in mobile. Or just don't hide it at all. 
What you did for desktop view, that's perfect. :)

**Comment by ardacebi** [comment]
@gurjyot I'm partial with you on the mobile description thing. I never tested the extension list on a mobile device but what kind of issues can it produce?

**Comment by CDK2020** [comment]
@gurjyot I think hiding it is the best option, It gets very squished with the 2 column look. Plus, there weren't descriptions before.

**Comment by CDK2020** [comment]
@gurjyot for your reference. here are the two changes proposed when squashed.

With descriptions removed:

image


Descriptions included: 
image

**Comment by gurjyot** [comment]
Even if we go with the first way, not having complete names upfront is just a mess. 
In this situation, I would like a single column implementation much better.

**Comment by datitisev** [comment]
For a smaller layout, i.e. <300px (I think that 992px is probably more for tablets), most likely one column would be preferable.

**Comment by CDK2020** [comment]
I've updated it to have 3 columns, but also reduce the number of columns if the screen can't fit the description, as well as completely hiding the description when it can't be fully shown (a screen less than 410px wide)

image

image

image

image

File: less/admin/ExtensionsPage.less [hunk 1]
```diff
@@ -96,3 +99,13 @@
   margin-left: -50px;
   position: absolute;
 }
+
+@media (max-width: 410px) {
```
Comment (franzliedke) [inline less/admin/ExtensionsPage.less]: Shouldn't this use on of our named breakpoints?

File: less/admin/ExtensionsPage.less [hunk 2]
```diff
@@ -96,3 +99,13 @@
...
+   .ExtensionListItem-description {
+        display: none;
+    }
+    .ExtensionListItem-version {
+        display: block;
+        margin-top: 8px;
```
Comment (franzliedke) [inline less/admin/ExtensionsPage.less]: Also, indentation seems wrong here. :stuck_out_tongue_winking_eye:

**Comment by tobscure** [comment]
I don't think it's a huge deal. In this case the 410px is preferable because that's the min-width of the columns?

**Comment by CDK2020** [comment]
@tobscure that's correct, this is the min-width of the columns. We can't use the breakpoints because those will hide the description before we need to.

**Comment by franzliedke** [comment]
In that case, you should create a new named breakpoint, to prevent magic numbers in the stylesheet.

**Comment by CDK2020** [comment]
@franzliedke after some research, there is no display on a phone currently in use with a width less than 410, in that case, I now favor removing the media constraint all together since it will foreseeably never be used

**Comment by datitisev** [comment]
@CDK2020 There may be some browsers that decrease the viewport size to make everything bigger, in which case you may want to hide the description.
It should probably remain included for those rare cases. It barely adds to the compiled CSS size.

**Comment by CDK2020** [comment]
@datitisev can you give an example of a browser doing that? How do you suggest we satisfy both your edge case as well as Franz's request?

**Comment by CDK2020** [comment]
@datitisev I feel like it is counter intuitive to make a breakpoint for just this one case, @franzliedke what part about `max-width` do you not like?

**Comment by franzliedke** [comment]
@CDK2020 The 410 can be considered a "magic number". When reading this stylesheet, you're left to wonder: Why this value and not any other? By using a breakpoint variable, we a) give it a name and b) place its definition with the other breakpoints, so that it cannot be forgotten when those change.

**Comment by tobscure** [comment]
I would've thought that the variable should be named for this specific use-case, eg. `@extension-list-column-width`. Probably don't need a variable for the media query, that can just go straight in ExtensionsPage.less. Is this what you had in mind @franzliedke?

**Comment by CDK2020** [comment]
@tobscure I've updated it per your request. Sorry for the trouble, I didn't quite understand what Franz was asking for.

**Comment by franzliedke** [comment]
Hmm, now you've added your very specific variable under the "Breakpoints" heading in variables.scss. That's not the right place.

If this number is specific for this page (as it seems to be), it's fine to have it in ExtensionsPage.less - but at least we give the concept a name, and cannot forget to change the breakpoint when we change the width of the column.

**Comment by CDK2020** [comment]
Hopefully that is the last commit needed!

**Comment by tobscure** [comment]
Thanks @CDK2020!

**Pull request merged**